cmd/sfncli/runner.go

```diff
@@ -165,7 +165,7 @@ func (t *TaskRunner) Process(ctx context.Context, args []string, input string) e
 }
 
 func (t *TaskRunner) handleSignals(ctx context.Context) {
-	sigChan := make(chan os.Signal)
+	sigChan := make(chan os.Signal, 1)
```

Comment:
Can you explain the context around this change?

Reply:
This makes sense as a best practice but the context should be documented in the PR

Reply:
building with go1.21 throws an error without this fix:

`cmd/sfncli/runner.go:169:2: misuse of unbuffered os.Signal channel as argument to signal.Notify`

Reply:
_If the channel were unbuffered (i.e., make(chan os.Signal)), and a signal arrived when the program was not in the select block, that signal would be lost. This is because sending to an unbuffered channel blocks until a receiver is ready, and there would be no receiver ready if the program is not in the select block.
So, a buffer of 1 makes sense here because it ensures that no signals are lost if they arrive at an inconvenient time. There’s generally no need for a larger buffer, because signals are not usually sent in rapid succession, and the program processes each signal immediately upon receiving it._